docs/plugin/custom-source.md

```diff
@@ -0,0 +1,232 @@
+
+## Prerequisites
+
+- Basic understanding of the Go language.
+- [Go](https://golang.org/doc/install) at least 1.18.
+  - See [go.mod](https://github.com/kubeshop/botkube/blob/main/go.mod#L1) for the recommended version used by Botkube team.
+- [GoReleaser](https://goreleaser.com/) at least 1.13.
+
+### Develop plugin business logic
+
+1. Create a source plugin directory:
+
+   ```bash
+   mkdir botkube-plugins && cd botkube-plugins
+   ```
+
+2. Initialize the Go module:
+
+   ```bash
+   go mod init botkube-plugins
+   ```
+
+3. Create the `main.go` file for the `ticker` source with the following template:
+
+   ```go
+   cat << EOF > main.go
+   package main
+
+   import (
+   	"context"
+   	"fmt"
+   	"time"
+
+   	"github.com/hashicorp/go-plugin"
+   	"github.com/kubeshop/botkube/pkg/api"
+   	"github.com/kubeshop/botkube/pkg/api/source"
+   	"gopkg.in/yaml.v3"
+   )
+
+   // Config holds executor configuration.
+   type Config struct {
+   	Interval time.Duration
+   }
+
+   // Ticker implements the Botkube source plugin interface.
+   type Ticker struct{}
+
+   func main() {
+   	source.Serve(map[string]plugin.Plugin{
+   		"ticker": &source.Plugin{
+   			Source: &Ticker{},
+   		},
+   	})
+   }
+   EOF
+   ```
+
+   This template code imports required packages and registers `Ticker` as the gRPC plugin. At this stage, the `Ticker` service doesn't implement the required [Protocol Buffers](https://github.com/kubeshop/botkube/blob/main/proto/source.proto) contract. We will add the required methods in the next steps.
+
+4. Download imported dependencies:
+
+   ```bash
+   go mod tidy
+   ```
+
+5. Add the required `Metadata` method:
+
+   ```go
+   // Metadata returns details about the Ticker plugin.
+   func (Ticker) Metadata(_ context.Context) (api.MetadataOutput, error) {
+   	return api.MetadataOutput{
+   		Version:     "0.1.0",
+   		Description: "Emits an event at a specified interval.",
+   	}, nil
+   }
+   ```
+
+   The `Metadata` method returns basic information about your plugin. This data is used when the plugin index is generated in an automated way. You will learn more about that in the next steps.
+
+6. Add the required `Stream` method:
+
+   ```go
+   // Stream sends an event after configured time duration.
+   func (Ticker) Stream(ctx context.Context, in source.StreamInput) (source.StreamOutput, error) {
+   	cfg, err := mergeConfigs(in.Configs)
+   	if err != nil {
+   		return source.StreamOutput{}, err
+   	}
+
+   	ticker := time.NewTicker(cfg.Interval)
+   	out := source.StreamOutput{
+   		Output: make(chan []byte),
+   	}
+
+   	go func() {
+   		for {
+   			select {
+   			case <-ctx.Done():
+   				ticker.Stop()
+   			case <-ticker.C:
+   				out.Output <- []byte("Ticker Event")
+   			}
+   		}
+   	}()
+
+   	return out, nil
+   }
+
+
+   // mergeConfigs merges all input configuration. In our case we don't have complex merge strategy,
```

Comment:
This somehow doesn't render properly:

https://plugin-source.botkube-docs.pages.dev/next/plugin/custom-source/

<img width="1068" alt="obraz" src="https://user-images.githubusercontent.com/7155799/212689629-827e9a7f-0ce7-4980-a363-f28f956bd204.png">

PTAL and fix it. Thanks!